File: solutions/python/grains/1/grains.py
def square(number):
    if number <= 0 or number > 64:
        raise ValueError("square must be between 1 and 64")
    if number == 1:
        return 1
    result = 1
    for n in range(2, number + 1):
        result *= 2
    return result


def total():
    result = 0
    for n in range(1, 65):
        result += square(n)
    return result

File: solutions/csharp/grains/1/Grains.cs
public static class Grains
{
    public static double Square(int n)
    {
        if (n <= 0 || n > 64) {
            throw new ArgumentOutOfRangeException();
        }
        
        return Math.Pow(2, n - 1);
    }

    public static double Total()
    {
        return Enumerable.Range(1, 64).Sum(Square);
    }
}